REST-API-Advanced/web/src/main/java/com/epam/esm/web/controller/GiftCertificateController.java

```diff
@@ -0,0 +1,75 @@
+package com.epam.esm.web.controller;
+
+import com.epam.esm.service.dto.GiftCertificateDto;
+import com.epam.esm.service.service.GiftCertificateService;
+import com.epam.esm.web.link.LinkAdder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/gift-certificates")
+public class GiftCertificateController {
+
+    private final GiftCertificateService giftCertificateService;
+    private final LinkAdder<GiftCertificateDto> giftCertificateDtoLinkAdder;
+
+    @Autowired
+    public GiftCertificateController(GiftCertificateService giftCertificateService, LinkAdder<GiftCertificateDto> giftCertificateDtoLinkAdder) {
+        this.giftCertificateService = giftCertificateService;
+        this.giftCertificateDtoLinkAdder = giftCertificateDtoLinkAdder;
+    }
+
+    @GetMapping(value = {"/{gift-certificate-id}", ""})
```

Comment:
gift-certificate-id is a Java variable, so use camel-case style. Additionally it can be simply named id, as it's clear from endpoint's path witch it relates for.